src/partition.ml

```diff
@@ -0,0 +1,94 @@
+open Encoding
+module H_contents = Irmin.Hash.Typed (Hash) (Contents)
+
+type entry = { name : string; kind : [ `Node | `Contents ]; hash : Hash.t }
+type enc_entry = { eencoding : string; entry : entry }
+type pointer = { index : int; hash : Hash.t }
+type enc_pointer = { pencoding : string; pointer : pointer }
+
+type vs =
+  | Empty
+  | Values of enc_entry list
+  | Tree of {
+      depth : int;
+      length : int;
+      pointers : (enc_pointer * enc_vs) list;
+    }
+
+and enc_vs = { vsencoding : string; vs : vs }
+
+let pre_hash t v =
+  let buf = Buffer.create 13 in
+  let pre_hash = Irmin.Type.(unstage (pre_hash t)) in
+  pre_hash v (Buffer.add_string buf);
+  Buffer.contents buf
+
+let compute_hash s = H_contents.hash (Bytes.of_string s)
+let pp_hash fmt h = Irmin.Type.(pp Encoding.Hash.t) fmt h
+let char_of_kind = function `Contents -> '\000' | `Node -> '\001'
+
+let to_enc_entry ({ name; kind; hash } as entry) =
+  let len = pre_hash Irmin.Type.int (String.length name) in
+  let eencoding =
+    Format.asprintf "%s%s%c%a" len name (char_of_kind kind) pp_hash hash
+  in
+  { eencoding; entry }
+
+let to_enc_pointer ({ index; hash } as pointer) =
+  let pencoding = Format.asprintf "%i%a" index pp_hash hash in
+  { pencoding; pointer }
+
+let pp_sep fmt () = Format.fprintf fmt ""
+let pp_int_to_leb128 fmt i = Format.fprintf fmt "%s" (pre_hash Irmin.Type.int i)
+
+let to_enc_vs vs =
+  let vsencoding =
+    match vs with
+    | Values l ->
+        Format.asprintf "\000\n%a"
```

Comment:
I don't think its `\n` but rather the length of `l`, no?

Reply:
Ok, this part of the documentation needs to be rewritten then because it's a bit mixed up.